5. 아래는 후보A의 선거구별 지지 현황표다. 후보A의 지지율이 선거구별로 차이가 있는지를 검정하라.
- 귀무가설, 대립가설을 설정하시오
- 소수점 2자리 반올림하여 검정통계량을 구하고, 95% 수준에서 연구가설 채택여부를 결정하라.

In [12]:
import pandas as pd
import numpy as np
df=pd.DataFrame({'ind':['지지', '지지x'],'1':[176, 124], '2':[193,107], '3':[159,141]})
df.set_index('ind', inplace=True)

(1) 귀무가설, 대립가설을 설정하시오
- H0 : 지지율과 선거구1,2,3은 독립이다(지지율은 모든 선거구에서 동일하다.)
- H1 : 지지율과 선거구1,2,3은 독립이 아니다(지지율이 모든 선거구에서 동일한 것은 아니다.)

In [13]:
from scipy.stats import chi2_contingency
chi, p, df, expect=chi2_contingency(df)
round(chi,2), p

(7.95, 0.01882272023214683)

검정통계량은 7.95

0.05보다 작으므로 귀무가설을 기각한다. 즉 지지율이 모든 선거구에서 동일한 것은 아니다.